src/main/java/seedu/address/model/Model.java

```diff
@@ -19,14 +20,27 @@
     void resetData(ReadOnlyTaskBook newData);
 
     /** Returns the TaskBook */
-    ReadOnlyTaskBook getAddressBook();
+    ReadOnlyTaskBook getTaskBook();
 
     /** Deletes the given task. */
     void deleteTask(Task target) throws TaskNotFoundException;
 
     /** Adds the given task */
     void addTask(Task task);
-
+    
+    //undo and redo
+    /** Undo prev action that modifies data**/
+    Command undo();
+    
+    /** reset stack of redoable actions when a non undo modifying data command is called**/
+    void resetRedoables();
```

Comment:
not possible, because

 @Override 
    public Command redo() throws EmptyStackException {
    	
    	TaskBook state = undoneStates.pop();
    	Command action = undoneCommands.pop();
    	recordStateBeforeChange(action);
    	resetData(state);
    	return action;
    }

here recordStateBeforeChange() is not followed by reset redo stack